server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java

```diff
@@ -505,33 +507,35 @@ protected final void connectToNodesAsync(ClusterState newClusterState, Runnable
         nodeConnectionsService.connectToNodes(newClusterState.nodes(), onCompletion);
     }
 
-    private void callClusterStateAppliers(ClusterChangedEvent clusterChangedEvent, StopWatch stopWatch) {
+    private void callClusterStateAppliers(ClusterChangedEvent clusterChangedEvent, Recorder stopWatch) {
         callClusterStateAppliers(clusterChangedEvent, stopWatch, highPriorityStateAppliers);
         callClusterStateAppliers(clusterChangedEvent, stopWatch, normalPriorityStateAppliers);
         callClusterStateAppliers(clusterChangedEvent, stopWatch, lowPriorityStateAppliers);
     }
 
-    private static void callClusterStateAppliers(ClusterChangedEvent clusterChangedEvent, StopWatch stopWatch,
+    private static void callClusterStateAppliers(ClusterChangedEvent clusterChangedEvent, Recorder stopWatch,
                                                  Collection<ClusterStateApplier> clusterStateAppliers) {
         for (ClusterStateApplier applier : clusterStateAppliers) {
             logger.trace("calling [{}] with change to version [{}]", applier, clusterChangedEvent.state().version());
-            try (Releasable ignored = stopWatch.timing("running applier [" + applier + "]")) {
+            final String name = applier.toString();
```

Comment:
I wonder if we should render nicer names here ... the output as rendered JSON is a little weird with all the `$` in them isn't it? (it's especially bothersome if the applier is a lambda and wouldn't even render to the same string on ever machine)

Reply:
Ah sorry this slipped right down my list, I need to get back to it soon.

We were planning to address the descriptions in a follow-up. I mean sure it's a bit weird to see the default `toString()` but it'd still tell us where to look at least.

Reply:
Yes, the plan is for each unique applier/listener to have a unique name.
I should have added as a comment to this PR.